rasa/core/policies/policy.py

```diff
@@ -195,6 +201,70 @@ def featurize_for_training(
 
         return state_features, label_ids, entity_tags
 
+    def prediction_states(
+        self,
+        tracker: DialogueStateTracker,
+        domain: Domain,
+        use_text_for_last_user_input: bool = False,
+        **kwargs: Any,
+    ) -> List[State]:
+        """Transforms tracker to states for prediction.
+
+        Args:
+            tracker: The tracker to be featurized.
+            domain: The Domain.
+            use_text_for_last_user_input: Indicates whether to use text or intent label
+                for featurizing last user input.
+
+        Returns:
+            A list of states.
+        """
+        return self.featurizer.prediction_states(
+            [tracker],
+            domain,
+            use_text_for_last_user_input=use_text_for_last_user_input,
+            ignore_rule_only_turns=self.supported_data() == SupportedData.ML_DATA,
+            rule_only_slots=kwargs.get("rule_only_slots"),
+            rule_only_loops=kwargs.get("rule_only_loops"),
+        )[0]
+
+    def featurize_for_prediction(
```

Comment:
this can be `protected`, no?
```suggestion
    def _featurize_for_prediction(
```